### PR TITLE
adapt EarlyStopping auto mode to auc

### PR DESCRIPTION
**Pull request by DachuanZhao**
Fix

[https://github.com/tensorflow/tensorflow/issues/44509](https://github.com/tensorflow/tensorflow/issues/44509)

[https://github.com/keras-team/keras/issues/15138](https://github.com/keras-team/keras/issues/15138)

### CODE REVIEW

File: keras/callbacks.py [hunk 1]
```diff
@@ -1782,7 +1782,7 @@ def __init__(self,
     elif mode == 'max':
       self.monitor_op = np.greater
     else:
-      if 'acc' in self.monitor:
+      if 'acc' in self.monitor or 'auc' in self.monitor:
```
Comment (fchollet) [inline keras/callbacks.py:1785]: prefer `or self.monitor == 'auc'` to avoid potential collisions

**Review by fchollet — CHANGES_REQUESTED** [review]
Thanks for the PR

File: keras/callbacks.py (hunk 1 above)
Comment (DachuanZhao) [reply keras/callbacks.py:1785]: The usual value of ```self.monitor``` is ```val_acc``` or ```val_auc``` . So it uses ```in``` here .

Comment (fchollet) [reply keras/callbacks.py:1785]: Then use `endswith`. Otherwise unrelated metrics that have the substring "auc" will get caught.

Comment (eli-osherovich) [reply keras/callbacks.py:1785]: my point was to use `endswith('_acc`)`

Comment (DachuanZhao) [reply keras/callbacks.py:1785]: How about 
```if self.monitor.endswith('acc') or self.monitor.endswith('accuracy') or self.monitor.endswith('auc')```
?

Comment (fchollet) [reply keras/callbacks.py:1785]: Yes, that works

Comment (DachuanZhao) [reply keras/callbacks.py]: done

File: keras/callbacks.py [hunk 2]
```diff
@@ -1782,7 +1782,7 @@ def __init__(self,
     elif mode == 'max':
       self.monitor_op = np.greater
     else:
-      if 'acc' in self.monitor:
+      if self.monitor.endswith('acc') or self.monitor.endswith('accuracy') or self.monitor.endswith('auc'):
```
Comment (fchollet) [inline keras/callbacks.py:1785]: Make sure the line in under 80 chars, e.g.

```
if (self.monitor.endswith('acc') or
    self.monitor.endswith('accuracy') or
    self.monitor.endswith('auc')):
```

Comment (DachuanZhao) [reply keras/callbacks.py]: Done. By the way , what python formatter does ```keras``` use ? ```yapf``` , ```black``` or something else ?

File: keras/callbacks.py [hunk 3]
```diff
@@ -1782,7 +1782,8 @@ def __init__(self,
     elif mode == 'max':
       self.monitor_op = np.greater
     else:
-      if 'acc' in self.monitor:
+      if (self.monitor.endswith('acc') or self.monitor.endswith('accuracy') or
+        self.monitor.endswith('auc')):
```
Comment (fchollet) [inline keras/callbacks.py:1786]: Fix indent

Comment (DachuanZhao) [reply keras/callbacks.py:1786]: I don't know fix what . The indent is two in ```keras``` code ...

Comment (fchollet) [reply keras/callbacks.py:1786]: Add two spaces.

Comment (DachuanZhao) [reply keras/callbacks.py]: Done.

**Review by fchollet — APPROVED** [review]
LGTM, thanks

**Pull request merged**